bench/wtperf/wtperf.c

```diff
@@ -1335,8 +1338,12 @@ checkpoint_worker(void *arg)
             if (wtperf->stop)
                 break;
         }
-        /* If the workers are done, don't bother with a final call. */
-        if (wtperf->stop)
+        /*
+         * If the workers are done, don't bother with a final call unless the flush tier worker
+         * needs to a final checkpoint to complete. The checkpoint thread keeps running as long as
+         * there is a flush tier worker thread running.
```

Comment:
This comment implies that we could keep doing checkpoints for a while if, for some reason, the flush thread keeps running.  In fact, the code will do one more checkpoint and then the checkpoint thread will leave this while loop and exit.  

I'm not sure if this is a faulty assumption in the code, or just an overly literal reading of the comment on my part.

Reply:
That makes sense.  I'm good.